gobblin-core/src/main/java/gobblin/writer/SimpleDataWriter.java

```diff
@@ -0,0 +1,175 @@
+/* (c) 2015 NerdWallet All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.writer;
+
+import com.google.common.base.Preconditions;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.util.ForkOperatorUtils;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+/**
+ * An implementation of {@link DataWriter} that writes bytes directly to HDFS.
+ *
+ * @author akshay@nerdwallet.com
+ */
+public class SimpleDataWriter extends BaseDataWriter<byte[]> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SimpleDataWriter.class);
+
+  private final FSDataOutputStream outputStream; // the output stream to the staging file
+  private final Byte recordDelimiter; // optional byte to place between each record write
+  private final boolean prependSize;
+
+  private int recordsWritten;
+  private int bytesWritten;
+  private boolean closed;
+
+  public SimpleDataWriter(State properties, String fileName, int numBranches, int branchId) throws IOException {
+    super(properties, fileName, numBranches, branchId);
+    String delim;
+    if ((delim = properties.getProp(ConfigurationKeys.SIMPLE_WRITER_DELIMITER, null)) == null || delim.length() == 0) {
+      this.recordDelimiter = null;
+    } else {
+      this.recordDelimiter = delim.getBytes()[0];
+    }
+
+    this.outputStream = this.fs.create(this.stagingFile, true);
+    this.prependSize = properties.getPropAsBoolean(ConfigurationKeys.SIMPLE_WRITER_PREPEND_SIZE, true);
+    this.recordsWritten = 0;
+    this.bytesWritten = 0;
+    this.closed = false;
+  }
+  /**
+   * Write a source record to the staging file
+   *
+   * @param record data record to write
+   * @throws java.io.IOException if there is anything wrong writing the record
+   */
+  @Override
+  public void write(byte[] record) throws IOException {
+    byte[] toWrite = record;
+    Preconditions.checkNotNull(record);
+    if (recordDelimiter != null) {
+      toWrite = Arrays.copyOf(record, record.length + 1);
+      toWrite[toWrite.length - 1] = recordDelimiter;
+    }
+    if (prependSize) {
+      Long recordSize = new Long(toWrite.length);
```

Comment:
What's the purpose of using `Long`? Why not just `long recordSize = toWrite.length`?